maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/test/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckinCommandTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.scm.provider.git.jgit.command.checkin;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.maven.scm.PlexusJUnit4TestSupport;
+import org.apache.maven.scm.ScmFileSet;
+import org.apache.maven.scm.ScmTestCase;
+import org.apache.maven.scm.command.checkin.CheckInScmResult;
+import org.apache.maven.scm.command.checkout.CheckOutScmResult;
+import org.apache.maven.scm.provider.git.GitScmTestUtils;
+import org.apache.maven.scm.repository.ScmRepository;
+import org.codehaus.plexus.util.FileUtils;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+
+public class JGitCheckinCommandTest extends ScmTestCase {
```

Comment:
I would prefer having this test as part of the Git TCK so that it is executed against both Git CLI and JGit.

Reply:
Good point. It turns out that the `RemoteRefUpdate.Status#UP_TO_DATE` equivalent for gitexe is considered ok as the `git push` command returns zero. Which seems questionable for maven-release. Nothing would be changed in the upstream.

Which is better: To have equal behavior for both implementations, or a successful, possibly empty release?

Reply:
I think a push which doesn't add any commits should never be an error (neither in JGit nor in GitExe). If there is an up2date check being necessary in maven-release-plugin that needs to be triggered separately. I would therefore appreciate to add the test to the TCK but assert that it is returning a success status.